src/protocolsupport/protocol/packet/middleimpl/clientbound/play/v_pe/PlayerInfo.java

```diff
@@ -45,6 +44,11 @@
 				return RecyclableEmptyList.get();
 			}
 		}
+
+		ClientBoundPacketData packetData = ClientBoundPacketData.create(PEPacketIDs.CRAFTING_DATA, ProtocolVersion.MINECRAFT_PE);
+		packetData.writeBytes(PECraftingManager.getInstance().getAllRecipes().array());
```

Comment:
Bad idea to use array, it may have bigger size then you have written to bytebuf

Reply:
what's my best alternative then? it was throwing recycle errors when i tried to cahce one ClientBountPacketData (obviously)

Reply:
Cache the byte array, not bytebuf. MiscSerializer::readAllBytes will read all bytes from ByteBuf properly.

Reply:
k, thanks